app/controllers/subscriptions_controller.rb

```diff
@@ -2,31 +2,36 @@ class SubscriptionsController < ApplicationController
   def create
     return render json: { id: 0 }, status: :created if smoke_test_address?
 
-    existing_subscription = nil
-    subscription = nil
+    subscription = send_email = status = nil
 
     Subscription.transaction do
+      deactivated_subscriber = subscriber.deactivated?
+      subscriber.activate! if deactivated_subscriber
+
       existing_subscription = Subscription.active.lock.find_by(
         subscriber: subscriber,
         subscriber_list: subscriber_list,
       )
 
-      existing_subscription.end(reason: :frequency_changed) if existing_subscription
+      create_new_subscription = existing_subscription.nil? ||
+        (frequency.to_s != existing_subscription.frequency.to_s)
 
-      subscriber.activate! if subscriber.deactivated?
+      if create_new_subscription
+        existing_subscription&.end(reason: :frequency_changed)
```

Comment:
Minor: I still find the `frequency_changed` stuff hard to follow, since you have to know about how `create_new_subscription` can be true. I think it might be easier to read this if we had another variable of `frequency_changed` and did the 'ending stuff' in a separate `if` block. How does that sound?

Reply:
I tried a few permutations but still can't come up with something better looking than this.